### ISSUE TITLE
[macOS] install_service_macos successfull, but jackett doesn't start at login and update don't work

### ISSUE THREAD

**Issue by sgarroo** [Jackett/Jackett#8329]
### Environment

MacOS 10.15.4 (19E287) Catalina | MacMiniServer (Late 2012) 

**Jackett Version**: v0.16.105

**Last Working Jackett Version**: none since the first i used

**Are you using a proxy or VPN?** [yes/no] not using VPN or proxy

### Description
Hello,
to use jackett i copy the folder uncompressed .tar in HD/Applications/Jackett and i start the ./Jackett inside and it works fine, to auto start the app I launch the executable as logon item, it opens a terminal window that i need to keep open to make it work.

If i try to install jackett as a service i get no error messages but simply it doesn't work.
If i remove the executable from the login items nothing happens, i tried to load unload and reload the service and the command works fine, saying that the service is already loaded.
to make things works i had to allow execution of unknown developer with the command
sudo spctl --master-disable , without this i had to accept every single time every component loaded by the script

i tryed also to search in the issue tab mixing answer from different thread but still nothing.

what is the best default installation path?
userhome or Application folder? 
is there a crafted .plist file i can load in the LaunchAgent folder modifying the execution path?
it would be awesome if everything can work by itself including auto-start and update, by now i am doing it manually every week because if I enable auto-update from the interface all file in the jackett folder will delete at the first attempt..

Thanks in advance for every advice.

Riccardo  
 

### Logged Error Messages
-- Running install_service_macos ->
```

MacMini:~ myuser$ /Applications/Jackett/install_service_macos ; exit;
Agent successfully installed and launched!
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]

```
-- Running JackettUpdater ->

```
MacMini:~ myuser$ /Applications/Jackett/JackettUpdater ; exit;
04-23 11:54:46 Info Jackett Updater v0.16.103 
04-23 11:54:46 Info Options "" 
04-23 11:54:47 Info Jackett variant: CoreMacOs 
04-23 11:54:47 Error JackettUpdater 0.16.103
Copyright (C) 2020 JackettUpdater

ERROR(S):
  Required option 'p, Path' is missing.

  -p, --Path     Required. Install location

  -t, --Type     Install type

  -a, --Args     Launch arguments

  --NoRestart    Don't restart after update

  --KillPids     PIDs which will be killed before (Windows) or after (Unix) the
                 update

  --StartTray    Indicates that the updater should start the tray icon

  --help         Display this help screen.

  --version      Display version information.

 
04-23 11:54:47 Error Failed to process update arguments! 
04-23 11:54:47 Error System.Linq.Enumerable+<ExceptIterator>d__80`1[CommandLine.Error] 

```
****keep waiting infinite time****

-- If i run it with -p path the result is this for every file in the folder and .. ->
```
MacMini:~ myuser$ sudo /Applications/Jackett/JackettUpdater -p /Applications/Jackett/
Password:
04-23 11:56:52 Info Jackett Updater v0.16.103 
04-23 11:56:53 Info Options "-p" "/Applications/Jackett/" 
04-23 11:56:53 Info Jackett variant: CoreMacOs 
04-23 11:56:53 Info Attempting to remove: /Applications/Jackett/DateTimeRoutines.pdb 
04-23 11:56:54 Info Deleted /Applications/Jackett/DateTimeRoutines.pdb 
04-23 11:56:54 Info Attempting to remove: /Applications/Jackett/JackettUpdater.pdb 
04-23 11:56:54 Info Deleted /Applications/Jackett/JackettUpdater.pdb 
04-23 11:56:55 Info Attempting to remove: /Applications/Jackett/jackett.pdb 
04-23 11:56:55 Info Deleted /Applications/Jackett/jackett.pdb 
04-23 11:56:55 Info Attempting to remove: /Applications/Jackett/Jackett.Common.pdb 
04-23 11:56:56 Info Deleted /Applications/Jackett/Jackett.Common.pdb 
04-23 11:56:56 Info Attempting to remove: /Applications/Jackett/jackett 
04-23 11:56:56 Info Deleted /Applications/Jackett/jackett 
04-23 11:56:56 Info Finding files in: /Applications/Jackett/ 
04-23 11:56:56 Info 817 update files found 
04-23 11:56:56 Info Attempting to copy AngleSharp.dll from source: /Applications/Jackett/AngleSharp.dll to destination: /Applications/Jackett/AngleSharp.dll 
04-23 11:56:56 Error System.IO.IOException: The process cannot access the file '/Applications/Jackett/AngleSharp.dll' because it is being used by another process.
   at System.IO.FileStream.Init(FileMode mode, FileShare share, String originalPath)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite)
   at System.IO.File.Copy(String sourceFileName, String destFileName, Boolean overwrite)
   at Jackett.Updater.Program.CopyUpdateFile(String jackettDestinationDirectory, String fullSourceFilePath, String updateSourceDirectory, Boolean previousAttemptFailed) System.IO.IOException: The process cannot access the file '/Applications/Jackett/AngleSharp.dll' because it is being used by another process.
   at System.IO.FileStream.Init(FileMode mode, FileShare share, String originalPath)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite)
   at System.IO.File.Copy(String sourceFileName, String destFileName, Boolean overwrite)
   at Jackett.Updater.Program.CopyUpdateFile(String jackettDestinationDirectory, String fullSourceFilePath, String updateSourceDirectory, Boolean previousAttemptFailed)
04-23 11:56:56 Info Attempting to copy AngleSharp.dll from source: /Applications/Jackett/AngleSharp.dll to destination: /Applications/Jackett/AngleSharp.dll 
04-23 11:56:56 Info The first attempt copying file: AngleSharp.dllfailed. Retrying and will delete old file first 
04-23 11:56:56 Info /Applications/Jackett/AngleSharp.dll was found 
04-23 11:56:57 Info Deleted /Applications/Jackett/AngleSharp.dll 
04-23 11:56:58 Error System.IO.FileNotFoundException: Could not find file '/Applications/Jackett/AngleSharp.dll'.
File name: '/Applications/Jackett/AngleSharp.dll'
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite)
   at System.IO.File.Copy(String sourceFileName, String destFileName, Boolean overwrite)
   at Jackett.Updater.Program.CopyUpdateFile(String jackettDestinationDirectory, String fullSourceFilePath, String updateSourceDirectory, Boolean previousAttemptFailed) System.IO.FileNotFoundException: Could not find file '/Applications/Jackett/AngleSharp.dll'.
File name: '/Applications/Jackett/AngleSharp.dll'
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite)
   at System.IO.File.Copy(String sourceFileName, String destFileName, Boolean overwrite)
   at Jackett.Updater.Program.CopyUpdateFile(String jackettDestinationDirectory, String fullSourceFilePath, String updateSourceDirectory, Boolean previousAttemptFailed)
04-23 11:56:58 Info Attempting to copy Autofac.dll from source: /Applications/Jackett/Autofac.dll to destination: /Applications/Jackett/Autofac.dll 
```
... and so on

-- this is my plist file that seems correct ->
```
MacMini:LaunchAgents myuser$ cat org.user.Jackett.plist 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>EnvironmentVariables</key>
    <dict>
        <key>PATH</key>
        <string>/usr/bin:/bin:/usr/sbin:/sbin</string>
    </dict>
    <key>KeepAlive</key>
    <true/>
    <key>Label</key>
    <string>org.user.Jackett</string>
    <key>ProgramArguments</key>
    <array>
        <string>/Applications/Jackett/jackett</string>
        <string>--NoRestart</string>
    </array>
    <key>RunAtLoad</key>
    <true/>
    <key>WorkingDirectory</key>
    <string>/Applications/Jackett</string>
</dict>
</plist>
```

**Comment by jonmcclay** [comment]
I am in this same situation. Jackett was fine after install. After restarting the process is not running and I don't know how to launch it again. I tried pasting

> launchctl load ~/Library/LaunchAgents/org.user.Jackett.plist

But that did not work...any tips would be appreciated

**Comment by garfield69** [comment]
ok no. 
I think you are running the installer as if that is the Jackett app.
revert the move to the applications folder and follow 
https://github.com/Jackett/Jackett#installation-on-macos

**Comment by sgarroo** [comment]
No, I assure you.. the install_service_macos isn't working (for me at least). I am using jackett executable as a login item or launch it manually from terminal and all it's working.
are there any specific permission / folder from where the installation has to be done?

**Comment by garfield69** [comment]
For my Mojave test image, I followed the installation notes and had no issues.
I found one ticket for Catalina, #8073 and it was resolved with a reboot.

Absolutely no idea what is going on with your setup.

I found one other open ticket for Catalina, read #5589. Perhaps this is affecting you too, albeit is silent mode (ie no popups just termination)?

**Comment by sgarroo** [comment]
Hi garfield69,

> ok no.
> I think you are running the installer as if that is the Jackett app.
> revert the move to the applications folder and follow
> https://github.com/Jackett/Jackett#installation-on-macos


to make it clear, if i move the folder from /Applications to User/Downloads and i run again the "install_service_macos", in the terminal i got this message:

`MacMini:~ myuser$ /Users/myuser/Downloads/Jackett/install_service_macos ; exit;
Agent successfully installed and launched!
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.`

if I check the .plist file in ~/Library/LaunchAgents/ it's updated with the new working directory:
`MacMini:~ myuser$ cat ~/Library/LaunchAgents/org.user.Jackett.plist 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>EnvironmentVariables</key>
    <dict>
        <key>PATH</key>
        <string>/usr/bin:/bin:/usr/sbin:/sbin</string>
    </dict>
    <key>KeepAlive</key>
    <true/>
    <key>Label</key>
    <string>org.user.Jackett</string>
    <key>ProgramArguments</key>
    <array>
        <string>/Users/myuser/Downloads/Jackett/jackett</string>
        <string>--NoRestart</string>
    </array>
    <key>RunAtLoad</key>
    <true/>
    <key>WorkingDirectory</key>
    <string>/Users/myuser/Downloads/Jackett</string>
</dict>
</plist>`

but there's no process named "jackett" running, never! if i move to another folder like "public" I have always the same result. I would like to run the app from /Applications/Jackett without having a terminal always open for that...
let me know if I'm missing something, but I know how to follow instructions and I know a bit what I am doing.

**Comment by sgarroo** [comment]
> For my Mojave test image, I followed the installation notes and had no issues.
> I found one ticket for Catalina, #8073 and it was resolved with a reboot.
> 
> Absolutely no idea what is going on with your setup.
> 
> I found one other open ticket for Catalina, read #5589. Perhaps this is affecting you too, albeit is silent mode (ie no popups just termination)?

I read those issues before opening this one, but i solved all the problems in the first installation allowing installation from any developer with this: "sudo spctl --master-disable" (so I don't have to accept all the dll and components to run unidentified apps or scripts, also my own)

**Comment by garfield69** [comment]
truthfully, I suspect you know more about your platform than I.
I use the mojave image occasionally, just to test Jackett on that platform.
So I cannot help you further.
@cadatoiva  @ngosang  anybody have some mac experience and can help diagnose this?

**Comment by sgarroo** [comment]
I have also installed the app on other macs running the same OS version and I have the same results. I'll try to downgrade one of those mac to Mojave or earlier and test it out... if I found out it works I'll look for the problem OS side. If I'll find out what prevents jackett to start from the launchd I'll tell you here... meanwhile any help is appreciated :)

**Comment by sgarroo** [comment]
ok.. I just downgraded my MacBookPro with Mojave and it works with 1 click... same thing for an older Mac with Sierra.
I'll look for problems with Catalina / LaunchAgents .plist execution.. Analyzing the plist I can see it runs for a second, than crashes and then a new PID for jackett is started, and loop. the error is "SIGABRT" seems something C++ related... 
With Catalina changes also the default shell.. I've already tried to add the path bin/zsh bin/bash to the .plist ENV VAR but still nothing..

**Comment by garfield69** [comment]
try to fish out more of that abend error... see if there is something in the event that identifies a module.
it is possible you are running into some kind of .net core 3.1 compatibility issue (a missing library or out of date library that needs upgrading or something along those lines).

**Comment by cadatoiva** [comment]
I have zero mac experience, as the last time I even touched a computer with MacOS was 2008. I will be of zero help in this issue.

**Comment by ngosang** [comment]
Sorry, I don't have macOS knowledge. You can try to search Catalina + .NET Core. Maybe https://github.com/dotnet/core/issues/3616

**Comment by jonmcclay** [comment]
> Hi garfield69,
> 
> > ok no.
> > I think you are running the installer as if that is the Jackett app.
> > revert the move to the applications folder and follow
> > https://github.com/Jackett/Jackett#installation-on-macos
> 
> to make it clear, if i move the folder from /Applications to User/Downloads and i run again the "install_service_macos", in the terminal i got this message:
> 
> ```
> MacMini:~ myuser$ /Users/myuser/Downloads/Jackett/install_service_macos ; exit;
> Agent successfully installed and launched!
> logout
> Saving session...
> ...copying shared history...
> ...saving history...truncating history files...
> ...completed.
> ```
> 
> if I check the .plist file in ~/Library/LaunchAgents/ it's updated with the new working directory:
> 
> ```
> MacMini:~ myuser$ cat ~/Library/LaunchAgents/org.user.Jackett.plist 
> <?xml version="1.0" encoding="UTF-8"?>
> <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
> <plist version="1.0">
> <dict>
>     <key>EnvironmentVariables</key>
>     <dict>
>         <key>PATH</key>
>         <string>/usr/bin:/bin:/usr/sbin:/sbin</string>
>     </dict>
>     <key>KeepAlive</key>
>     <true/>
>     <key>Label</key>
>     <string>org.user.Jackett</string>
>     <key>ProgramArguments</key>
>     <array>
>         <string>/Users/myuser/Downloads/Jackett/jackett</string>
>         <string>--NoRestart</string>
>     </array>
>     <key>RunAtLoad</key>
>     <true/>
>     <key>WorkingDirectory</key>
>     <string>/Users/myuser/Downloads/Jackett</string>
> </dict>
> </plist>
> ```
> 
> but there's no process named "jackett" running, never! if i move to another folder like "public" I have always the same result. I would like to run the app from /Applications/Jackett without having a terminal always open for that...
> let me know if I'm missing something, but I know how to follow instructions and I know a bit what I am doing.

I have the jackett folder in the applications folder. That’s where I have success until a reboot

**Comment by narmbrister** [comment]
I am running Jackett on Catalina (same versions) as a service and I haven't had any problems. I have it installed in a different folder, but even when I installed a fresh copy in my Applications folder as per your steps it still worked normally. Same plist file and everything. I have configured my firewall to allow incoming connections for the jackett binary file, but I doubt that has anything to do with your problem

Gatekeeper has gone a bit funny in Catalina, so maybe you could try removing the quarantine flags instead of disabling it entirely (probably a good idea anyway). You can do that with the command "xattr -rd com.apple.quarantine /path/to/Jackett". The "r" flag makes it recursive so you won't have to repeat it for every file individually.

Hope that helps!

**Comment by sgarroo** [comment]
Tried but without success all of these things:
"xattr -rd" for jackett working folder (but if I start manually the jackett executable has always worked)
made some tests with .plist file, the only one I found working is to run a .sh with the path/to/Jackett/jackett file with launchd (another way  and nothing more than put the file in "Login items") jackett auto update and service restart after failure won't work anyway..

**Comment by sgarroo** [comment]
``launchctl list org.user.Jackett
{
	"LimitLoadToSessionType" = "Aqua";
	"Label" = "org.user.Jackett";
	"OnDemand" = false;
	"LastExitStatus" = 6;
	"Program" = "/Applications/Jackett/jackett";
	"ProgramArguments" = (
		"/Applications/Jackett/jackett";
		"--NoRestart";
	);
};
``
`launchctl print gui/501/org.user.Jackett
org.user.Jackett = {
	active count = 0
	copy count = 0
	one shot = 0
	path = /Users/sgarroo/Library/LaunchAgents/org.user.Jackett.plist
	state = pending

	program = /Applications/Jackett/jackett
	arguments = {
		/Applications/Jackett/jackett
		--NoRestart
	}

	working directory = /Applications/Jackett

	inherited environment = {
		SSH_AUTH_SOCK => /private/tmp/com.apple.launchd.urNq76qdGB/Listeners
	}

	default environment = {
		PATH => /usr/bin:/bin:/usr/sbin:/sbin
	}

	environment = {
		PATH => /usr/bin:/bin:/usr/sbin:/sbin
		XPC_SERVICE_NAME => org.user.Jackett
	}

	domain = com.apple.xpc.launchd.user.domain.501.100006.Aqua
	asid = 100006
	minimum runtime = 10
	exit timeout = 5
	runs = 8275
	successive crashes = 8275
	last terminating signal = Abort trap: 6

	event triggers = {
	}

	endpoints = {
	}

	dynamic endpoints = {
	}

	pid-local endpoints = {
	}

	instance-specific endpoints = {
	}

	event channels = {
	}

	sockets = {
	}

	instances = {
	}

	spawn type = daemon
	spawn role = (null)
	jetsam priority = 3
	jetsam memory limit (active) = (unlimited)
	jetsam memory limit (inactive) = (unlimited)
	jetsamproperties category = daemon
	submitted job. ignore execute allowed
	jetsam thread limit = 32
	cpumon = default

	properties = {
		partial import = 0
		launchd bundle = 0
		xpc bundle = 0
		keepalive = 1
		runatload = 1
		low priority i/o = 0
		low priority background i/o = 0
		dataless file mode = 0
		legacy timer behavior = 0
		exception handler = 0
		multiple instances = 0
		supports transactions = 0
		supports pressured exit = 0
		supports idle hysteresis = 0
		enter kdp before kill = 0
		wait for debugger = 0
		app = 0
		system app = 0
		creates session = 0
		inetd-compatible = 0
		inetd listener = 0
		abandon process group = 0
		one-shot = 0
		event monitor = 0
		penalty box = 0
		pended non-demand spawn = 0
		role account = 0
		launch only once = 0
		system support = 0
		app-like = 0
		inferred program = 1
		joins gui session = 0
		joins host session = 0
		parameterized sandbox = 0
		resolve program = 0
		abandon coalition = 0
		high bits aslr = 0
		extension = 0
		nano allocator = 0
		no initgroups = 0
		start on fs mount = 0
		endpoints initialized = 1
		is copy = 0
		disallow all lookups = 0
		system service = 0
		protected by submitter = 0
	}
}`


`cd ~/Library/LaunchAgents; ls -la
total 32
drwxr-xr-x@  5 sgarroo  staff   160 Jul 13 11:34 .
drwx------+ 75 sgarroo  staff  2400 May 11 12:56 ..
-rw-r--r--@  1 sgarroo  staff  6148 Nov 30  2019 .DS_Store
-rw-r--r--   1 sgarroo  staff   687 Jun 22  2019 com.adobe.GC.Invoker-1.0.plist
-rw-r--r--   1 sgarroo  staff   684 Jul 13 11:34 org.user.Jackett.plist`


sorry for the delay in the answer!

**Comment by punkhop** [comment]
Similar issue. Using 10.15.6. Install reported fine, but service isn't running. Tried in Applications and Downloads folders.

Also tried running manually with ./jackett and get this error:

Franks-MacBook-Air:Jackett frankguercio$ ./jackett
09-17 17:16:22 Info Starting Jackett v0.16.1231.0
Unhandled exception. System.Exception: Could not create settings directory. The file
'/Applications/Jackett/Jackett' already exists.
at Jackett.Common.Services.ConfigurationService.CreateOrMigrateSettings() in
/Users/runner/work/1/s/src/Jackett.Common/Services/ConfigurationService.cs:line 52
at Jackett.Server.Program.Main(String[] args) in
/Users/runner/work/1/s/src/Jackett.Server/Program.cs:line 90
Abort trap: 6

**Comment by deecypher** [comment]
Not to hijack this thread, but I'm having the exact same problem. However, I'm on the latest OS 12.3.1. I can manually start jackett but running as a service never works on restart. Anyone out there solve this issue?

**Comment by ilike2burnthing** [comment]
Install and uninstall scripts have been updated by https://github.com/Jackett/Jackett/pull/13626, available in [v0.20.2080](https://github.com/Jackett/Jackett/releases/tag/v0.20.2080). Please check if your issue is resolved.
